### PR TITLE
feat: adds new fields to template

### LINKED ISSUE #93: Require project name for all jobs
Acceptance criteria
- [ ] column added to template
- [ ] column should have a dropdown that mirrors the project smartsheet
- [ ] data_description.json contains the submitted project name
- [ ] funding information is pulled from aind-metadata-service using this field

### LINKED ISSUE #94: Add codeocean-process-capsule-id to BasicUploadJobConfigs (or other schema)
I would like to be able to trigger a pipeline through aind-data-transfer-service. It would be cool if I could pass in the codeocean-process-capsule-id as an extra config parameter to the current job schema.

### PR DESCRIPTION
**Pull request by jtyoung84**
Closes #94 
Closes #93 

- [x] Needs https://github.com/AllenNeuralDynamics/aind-metadata-service/pull/225 to be merged first
- Adds column for processor_full_name
- Adds column for project_name
- Adds column for optional process_capsule_id
- Modifies job_template download to pull project_names from aind-metadata-service

### CODE REVIEW

**Review by arielleleon — APPROVED** [review]
Looks good. On my end, I will need to update how to map current LIMS project codes to the correct project name. I am working with PIs on that right now. Thanks for doing this work so fast!

File: tests/test_server.py [hunk 1]
```diff
@@ -534,44 +544,91 @@ def test_jobs_failure(self, mock_get: MagicMock):
...
+
+        mock_response = Response()
+        mock_response.status_code = 200
+        mock_project_names = ["Ephys Platform", "Behavior Platform"]
+        mock_response._content = json.dumps(
+            {"data": mock_project_names}
+        ).encode("utf-8")
+        mock_get.return_value = mock_response
+
+        # mock_create_template.return_value = BytesIO(b"mock_template_stream")
```
Comment (jtyoung84) [inline tests/test_server.py:559]: delete this

File: src/aind_data_transfer_service/configs/job_upload_template.py [hunk 2]
```diff
@@ -57,42 +80,57 @@ class JobUploadTemplate:
...
+            {
+                "name": "platform",
+                "type": "list",
+                "options": [p().abbreviation for p in Platform._ALL],
+                "column_indexes": [self.HEADERS.index("platform")],
+            },
+            {
+                "name": "project_name",
+                "type": "list",
+                "options": [p for p in self.project_names],
```
Comment (helen-m-lin) [inline src/aind_data_transfer_service/configs/job_upload_template.py:103]: It looks like Excel can only store a max of 255 chars for Data validators defined directly as a list. Is there a way we can shorted these? Or, we may need to write the project names into another worksheet in the same workbook and use a formula to pull them.

File: tests/resources/job_upload_template.xlsx
Comment (helen-m-lin) [inline tests/resources/job_upload_template.xlsx:1]: When I open this locally, Excel removes `project_name` validation dropdown.

**Review by helen-m-lin — CHANGES_REQUESTED** [review]
Changes look good. Can we update the usage instructions and example in the README and add `AIND_PROJECT_NAMES_URL`?

Lastly, I am unable to see the `project_names` validation dropdown in the downloaded template. Will discuss offline.

File: src/aind_data_transfer_service/configs/job_upload_template.py (hunk 2 above)
Comment (jtyoung84) [reply src/aind_data_transfer_service/configs/job_upload_template.py]: Removed the validator for the time being

File: tests/resources/job_upload_template.xlsx
Comment (jtyoung84) [reply tests/resources/job_upload_template.xlsx:1]: Removed the validator from the template

File: src/aind_data_transfer_service/templates/index.html [hunk 3]
```diff
@@ -47,6 +47,7 @@
         <a href="/">Submit Jobs</a> |
         <a href="/jobs">Job Status</a> |
         <a title="Download job template as .xslx" href= "/api/job_upload_template" download>Job Submit Template</a> |
+        <a title="List of project names" href= "http://aind-metadata-service/project_names" >Project Names</a> |
```
Comment (helen-m-lin) [inline src/aind_data_transfer_service/templates/index.html:50]: nit: open in new tab?
```suggestion
        <a title="List of project names" href= "http://aind-metadata-service/project_names" target="_blank" >Project Names</a> |
```

File: src/aind_data_transfer_service/templates/job_status.html [hunk 4]
```diff
@@ -26,7 +26,9 @@
     <nav>
         <a href="/">Submit Jobs</a> |
         <a href="/jobs">Job Status</a> |
-        <a title="Download job template as .xslx" href= "/api/job_upload_template" download>Job Submit Template</a>
+        <a title="Download job template as .xslx" href= "/api/job_upload_template" download>Job Submit Template</a> |
+        <a title="List of project names" href= "http://aind-metadata-service/project_names" >Project Names</a> |
```
Comment (helen-m-lin) [inline src/aind_data_transfer_service/templates/job_status.html:30]: nit: open in new tab?

```suggestion
        <a title="List of project names" href= "http://aind-metadata-service/project_names" target="_blank" >Project Names</a> |
```

**Pull request merged**